### PR TITLE
ARROW-12202: [Rust] WIP Fix SEGFAULT/ SIGILL in child-data ffi

### PR DESCRIPTION
**Pull request by ritchie46**
Continue #9778

### CODE REVIEW

File: rust/arrow/src/ffi.rs [hunk 1]
```diff
@@ -135,13 +140,14 @@ impl FFI_ArrowSchema {
...
+        let flags = if nullable { FLAG_NULLABLE } else { FLAG_NONE };
 
         let private_data = Box::new(SchemaPrivateData { children });
         // <https://arrow.apache.org/docs/format/CDataInterface.html#c.ArrowSchema>
         FFI_ArrowSchema {
             format: CString::new(format).unwrap().into_raw(),
-            // For child data a non null string is expected and is called item
+            // TODO: get the field name for general nested data
+            // For List child data a non null string is expected and is called item
             name: CString::new("item").unwrap().into_raw(),
```
Comment (pitrou) [inline rust/arrow/src/ffi.rs:151]: This should be released in `release_schema`, like `format` already is.

File: rust/arrow/src/ffi.rs [hunk 2]
```diff
@@ -395,6 +397,12 @@ unsafe extern "C" fn release_array(array: *mut FFI_ArrowArray) {
     // take ownership of `private_data`, therefore dropping it
     Box::from_raw(array.private_data as *mut PrivateData);
 
+    // release children
+    for i in 0..array.n_children {
+        let child = *array.children.add(i as usize);
+        release_array(child);
```
Comment (pitrou) [inline rust/arrow/src/ffi.rs:403]: You should call the release callback stored on `child`, not `release_array` directly.

Comment (pitrou) [reply rust/arrow/src/ffi.rs:403]: Note you must do the same thing in `release_schema` as well.

**Comment by alamb** [comment]
The Apache Arrow Rust community is moving the Rust implementation into its own dedicated github repositories [arrow-rs](https://github.com/apache/arrow-rs) and [arrow-datafusion](https://github.com/apache/arrow-datafusion). It is likely we will not merge this PR into this repository

Please see the [mailing-list](https://lists.apache.org/thread.html/rce7c61cb3f703367dc00984530016e3fcb828e5a8035655fbcccf862%40%3Cdev.arrow.apache.org%3E) thread for more details

We expect the process to take a few days and will follow up with a migration plan for the in-flight PRs.

**Comment by nevi-me** [comment]
Reopened at https://github.com/apache/arrow-rs/pull/21

**Pull request closed without merging**